src/core_plugins/timelion/public/directives/timelion_expression_input_helpers.js

```diff
@@ -0,0 +1,105 @@
+import _ from 'lodash';
+
+export class FunctionSuggestions {
+  constructor() {
+    this.reset();
+  }
+
+  reset() {
+    this.index = 0;
+    this.list = [];
+    this.isVisible = false;
+  }
+
+  setList(list) {
+    this.list = list;
+
+    // We may get a shorter list than the one we have now, so we need to make sure our index doesn't
+    // fall outside of the new list's range.
+    this.index = Math.max(0, Math.min(this.index, this.list.length - 1));
+  }
+
+  getCount() {
+    return this.list.length;
+  }
+
+  isEmpty() {
+    return this.list.length === 0;
+  }
+
+  show() {
+    this.isVisible = true;
+  }
+
+  hide() {
+    this.isVisible = false;
+  }
+
+  stepForward() {
+    if (this.index > 0) {
+      this.index -= 1;
+    }
+  }
+
+  stepBackward() {
+    if (this.index < this.list.length - 1) {
+      this.index += 1;
+    }
+  }
+}
+
+export function suggest(expression, caretOffset, functionList, Parser) {
```

Comment:
could you fix this function so it also shows the suggestion after you start writing your arguments in ?
`.es` ... shows me the doc for es function with all the arguments
after i type `.es(` the arguments disappear which is supper annoying.

i understand if you want to do it in separate pr but it might be a small change ....
maybe storing last valid caret position and in just showing the doc for that ?